src/GlobalState.lua

```diff
@@ -510,4 +510,29 @@ function M.update_queue(update_entity)
   Queue.swap_random_to_front(global.mod.scan_queue, global.mod.rand)
 end
 
+-- translate a tile name to the item name ("stone-path" => "stone-brick")
+function M.resolve_name(name)
+  if game.item_prototypes[name] ~= nil or game.fluid_prototypes[name] ~= nil then
+    return name
+  end
+
+  local prot = game.tile_prototypes[name]
+  if prot ~= nil then
+    local mp = prot.mineable_properties
+    if mp.minable and #mp.products == 1 then
+      return mp.products[1].name
+    end
+  end
+
+  -- FIXME: figure out how to not hard-code this
```

Comment:
Do you have any ideas for how to not hard-code this information?

Reply:
Nope. I haven't found "curved-rail" in a runtime prototype structure.
It is at data.raw["curved-rail"]["curved-rail"] in the data stage. Not promising.